### PR TITLE
Refactored PartnersProxyAPI and ScriptServiceAPI (now PartnersProxyBypassAPI) in ScriptService

### PR DESCRIPTION
**Pull request by SpencerHaan**
### WHY are these changes introduced?

This is an initial refactor to make it easier to do further refactoring in the ScriptService class. The goal is to get rid of `MakeRequest` and move `UploadScript` out. These two proxy classes represented a bit of a barrier to this, so they needed to be refactored in order to move the proxy related code out of `MakeRequest`.

### WHAT is this pull request doing?

Instead of using the query method on the two API based subclasses statically, I've instead made them instance based in order to create a common interface between them. This allows us to swap them out instead of having to have `MakeRequest` perform this distinction by wrapping the calls to either.

### Update checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).

### CODE REVIEW

File: lib/project_types/script/layers/infrastructure/script_service.rb [hunk 1]
```diff
@@ -66,34 +66,31 @@ def push(
...
 
-          def self.query(ctx, query_name, api_key: nil, variables: {})
-            api_client(ctx, api_key).query(query_name, variables: variables)
-          end
-
-          def self.api_client(ctx, api_key)
+          def initialize(ctx, api_key)
             instance_url = spin_instance_url || LOCAL_INSTANCE_URL
-            new(
+            super(
```
[the remark this answers is not in the document]
Comment (maximebedard) [reply lib/project_types/script/layers/infrastructure/script_service.rb:83]: I believe they aren't the same value, but I could be wrong. At the time when I was originally writing this it wasn't possible to override the authentication tokens.

File: lib/project_types/script/layers/infrastructure/script_service.rb [hunk 2]
```diff
@@ -166,6 +139,43 @@ def error_code(errors)
...
           end
+
+          class ShimAPI < ShopifyCli::PartnersAPI
+            def query(query_name, variables: {})
+              variables[:query] = load_query(query_name)
+              super("script_service_proxy", variables: variables)
+            end
+          end
+          private_constant(:ShimAPI)
+        end
```
[the remark this answers is not in the document]
Comment (ErinRenNumber1) [reply lib/project_types/script/layers/infrastructure/script_service.rb:150]: Unfortunate indeed!

File: lib/project_types/script/layers/infrastructure/script_service.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ErinRenNumber1) [reply lib/project_types/script/layers/infrastructure/script_service.rb:83]: I will leave it up to you to check.  😄

File: lib/project_types/script/layers/infrastructure/script_service.rb [hunk 3]
```diff
@@ -66,34 +66,31 @@ def push(
...
+          response = MakeRequest.new(ctx, api_key).call(
             query_name: query_name,
-            api_key: api_key,
             variables: variables
           )
           response["data"]["appScripts"]
         end
 
-        class ScriptServiceAPI < ShopifyCli::API
+        class PartnersProxyBypassAPI < ShopifyCli::API
```
[the remark this answers is not in the document]
Comment (ErinRenNumber1) [reply lib/project_types/script/layers/infrastructure/script_service.rb:76]: I will leave it up to you.  I am not too concerned here.

**Review by ErinRenNumber1 — APPROVED** [review]

File: lib/project_types/script/layers/infrastructure/script_service.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (SpencerHaan) [reply lib/project_types/script/layers/infrastructure/script_service.rb]: Ended up just keeping it as `ScriptServiceAPI`

**Pull request merged**